expression/builtin_math.go

```diff
@@ -1074,19 +1074,33 @@ type piFunctionClass struct {
 }
 
 func (c *piFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinPISig{newBaseBuiltinFunc(args, ctx)}
-	return sig.setSelf(sig), errors.Trace(c.verifyArgs(args))
+	var (
+		bf  baseBuiltinFunc
+		sig builtinFunc
+		err error
+	)
+
+	if err = c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	if bf, err = newBaseBuiltinFuncWithTp(args, ctx, tpReal); err != nil {
+		return nil, errors.Trace(err)
+	}
+
```

Comment:
@zz-jason 
It seems we set `Flen to 17` and `Decimal to 15` will be better,
TiDB cannot handle `select pi() + 0.0000` now,
so keep the `flen` and `decimal` consistent with the certain value may be a tradeoff.

@0x04C2  Sorry for making you modifying once again.